<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Decorate truncated sequences — geom_seq_break • gggenomes</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><script src="../extra.js"></script><meta property="og:title" content="Decorate truncated sequences — geom_seq_break"><meta property="og:description" content="geom_seq_break() adds decorations to the ends of truncated sequences. These
could arise from zooming onto sequence loci with focus(), or manually
annotating sequences with start &amp;gt; 1 and/or end &amp;lt; length."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.10.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thackl/gggenomes/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Decorate truncated sequences</h1>
    <small class="dont-index">Source: <a href="https://github.com/thackl/gggenomes/blob/HEAD/R/geom_seq_break.R" class="external-link"><code>R/geom_seq_break.R</code></a></small>
    <div class="hidden name"><code>geom_seq_break.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>geom_seq_break()</code> adds decorations to the ends of truncated sequences. These
could arise from zooming onto sequence loci with <code><a href="focus.html">focus()</a></code>, or manually
annotating sequences with start &gt; 1 and/or end &lt; length.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geom_seq_break</span><span class="op">(</span></span>
<span>  mapping_start <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mapping_end <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data_start <span class="op">=</span> <span class="fu"><a href="pull_track.html">seqs</a></span><span class="op">(</span><span class="va">start</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>  data_end <span class="op">=</span> <span class="fu"><a href="pull_track.html">seqs</a></span><span class="op">(</span><span class="va">end</span> <span class="op">&lt;</span> <span class="va">length</span><span class="op">)</span>,</span>
<span>  label <span class="op">=</span> <span class="st">"/"</span>,</span>
<span>  size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  hjust <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>  family <span class="op">=</span> <span class="st">"bold"</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data_start</dt>
<dd><p>seq_layout of sequences for which to decorate the start.
default: <code>seqs(start &gt;1)</code></p></dd>


<dt>data_end</dt>
<dd><p>seq_layout of sequences for which to decorate the end.
default: <code>seqs(end &lt; length)</code></p></dd>


<dt>label</dt>
<dd><p>the character to decorate ends with. Provide two values for
different start and end decorations, e.g. <code>label=c("]", "[")</code>.</p></dd>


<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, either as a <code>ggproto</code> <code>Geom</code> subclass or as a string naming the
stat stripped of the <code>stat_</code> prefix (e.g. <code>"count"</code> rather than
<code>"stat_count"</code>)</p></dd>


<dt>na.rm</dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>


<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>


<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>


<dt>...</dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># decorate breaks created with focus()</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">emale_genes</span>, <span class="va">emale_seqs</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="focus.html">focus</a></span><span class="op">(</span>.expand<span class="op">=</span><span class="fl">1e3</span>, .max_dist <span class="op">=</span> <span class="fl">1e3</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_seq_break</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Showing 10 loci with the following size distribution</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> min: 3853</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> q25: 9284</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> med: 12695</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> q75: 18007</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> max: 21311</span>
<span class="r-plt img"><img src="geom_seq_break-1.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># customize decorations</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">emale_genes</span>, <span class="va">emale_seqs</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="focus.html">focus</a></span><span class="op">(</span>.expand<span class="op">=</span><span class="fl">1e3</span>, .max_dist <span class="op">=</span> <span class="fl">1e3</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_seq_break</span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"["</span>, <span class="st">"]"</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"#1b9e77"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Showing 10 loci with the following size distribution</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> min: 3853</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> q25: 9284</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> med: 12695</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> q75: 18007</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> max: 21311</span>
<span class="r-plt img"><img src="geom_seq_break-2.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># decorate manually truncated sequences</span></span></span>
<span class="r-in"><span><span class="va">s0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tribble.html" class="external-link">tribble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="co"># start/end define regions, i.e. truncated contigs</span></span></span>
<span class="r-in"><span>  <span class="op">~</span><span class="va">bin_id</span>, <span class="op">~</span><span class="va">seq_id</span>, <span class="op">~</span><span class="va">length</span>, <span class="op">~</span><span class="va">start</span>, <span class="op">~</span><span class="va">end</span>,</span></span>
<span class="r-in"><span>  <span class="st">"complete_genome"</span>, <span class="st">"chromosome_1_long_trunc_2side"</span>, <span class="fl">1e5</span>, <span class="fl">1e4</span>, <span class="fl">2.1e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"fragmented_assembly"</span>, <span class="st">"contig_1_trunc_1side"</span>, <span class="fl">1.3e4</span>, <span class="fl">.9e4</span>, <span class="fl">1.3e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"fragmented_assembly"</span>, <span class="st">"contig_2_short_complete"</span>, <span class="fl">0.3e4</span>, <span class="fl">1</span>, <span class="fl">0.3e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"fragmented_assembly"</span>, <span class="st">"contig_3_trunc_2sides"</span>, <span class="fl">2e4</span>, <span class="fl">1e4</span>, <span class="fl">1.4e4</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">l0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tribble.html" class="external-link">tribble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="op">~</span><span class="va">seq_id</span>, <span class="op">~</span><span class="va">start</span>, <span class="op">~</span><span class="va">end</span>, <span class="op">~</span><span class="va">seq_id2</span>, <span class="op">~</span><span class="va">start2</span>, <span class="op">~</span><span class="va">end2</span>,</span></span>
<span class="r-in"><span>  <span class="st">"chromosome_1_long_trunc_2side"</span>, <span class="fl">1.1e4</span>, <span class="fl">1.4e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"contig_1_trunc_1side"</span>, <span class="fl">1e4</span>, <span class="fl">1.3e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"chromosome_1_long_trunc_2side"</span>, <span class="fl">1.4e4</span>, <span class="fl">1.7e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"contig_2_short_complete"</span>, <span class="fl">1</span>, <span class="fl">0.3e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"chromosome_1_long_trunc_2side"</span>, <span class="fl">1.7e4</span>, <span class="fl">2e4</span>,</span></span>
<span class="r-in"><span>  <span class="st">"contig_3_trunc_2sides"</span>, <span class="fl">1e4</span>, <span class="fl">1.3e4</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="gggenomes.html">gggenomes</a></span><span class="op">(</span>seqs<span class="op">=</span><span class="va">s0</span>, links<span class="op">=</span><span class="va">l0</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="geom_link.html">geom_link</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span>nudge_y<span class="op">=</span><span class="op">-</span><span class="fl">.05</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_seq_break</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_seq_break-3.png" alt="" width="960" height="480"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thomas Hackl, Markus J. Ankenbrand, Bart van Adrichem.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

