<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add text to genes, features, etc. — geom_feat_text • gggenomes</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><script src="../extra.js"></script><meta property="og:title" content="Add text to genes, features, etc. — geom_feat_text"><meta property="og:description" content="The functions below are useful for labeling features/genes in plots.
Users have to call on aes(label = ...) or (label = ...) to define label's text
Based on the function, the label will be placed at a specific location:
geom_..._text() will plot text in the middle of the feature.
geom_..._tag()  will plot text on top of the feature, with a 45 degree angle.
geom_..._note() will plot text under the feature at the left side.


The ... can be either replaced with feat or gene depending on which
track the user wants to label.
With arguments such as hjust, vjust, angle, and nudge_y, the user
can also manually change the position of the text."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thackl/gggenomes/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add text to genes, features, etc.</h1>
    <small class="dont-index">Source: <a href="https://github.com/thackl/gggenomes/blob/HEAD/R/geom_gene_text.R" class="external-link"><code>R/geom_gene_text.R</code></a></small>
    <div class="hidden name"><code>geom_feat_text.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The functions below are useful for labeling features/genes in plots.
Users have to call on <code>aes(label = ...)</code> or <code>(label = ...) </code>to define label's text
Based on the function, the label will be placed at a specific location:</p><ul><li><p><code>geom_..._text()</code> will plot <strong>text in the middle of the feature</strong>.</p></li>
<li><p><code>geom_..._tag()</code>  will plot <strong>text on top of the feature, with a 45 degree angle</strong>.</p></li>
<li><p><code>geom_..._note()</code> will plot <strong>text under the feature at the left side</strong>.</p></li>
</ul><p><em>The <code>...</code> can be either replaced with <code>feat</code> or <code>gene</code> depending on which</em>
<em>track the user wants to label.</em></p>
<p>With arguments such as <code>hjust</code>, <code>vjust</code>, <code>angle</code>, and <code>nudge_y</code>, the user
can also manually change the position of the text.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geom_feat_text</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">feats</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geom_feat_tag</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">feats</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  vjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>  xjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  strandwise <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geom_feat_note</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">feats</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  vjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="op">-</span><span class="fl">0.03</span>,</span>
<span>  xjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  strandwise <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geom_gene_text</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geom_gene_tag</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  vjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>  xjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  strandwise <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geom_gene_note</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="fu"><a href="pull_track.html">genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  vjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="op">-</span><span class="fl">0.03</span>,</span>
<span>  xjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  strandwise <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  parse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  check_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mapping</dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p></dd>


<dt>data</dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>


<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, either as a <code>ggproto</code> <code>Geom</code> subclass or as a string naming the
stat stripped of the <code>stat_</code> prefix (e.g. <code>"count"</code> rather than
<code>"stat_count"</code>)</p></dd>


<dt>position</dt>
<dd><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function. Cannot be jointly specified with
<code>nudge_x</code> or <code>nudge_y</code>.</p></dd>


<dt>...</dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p></dd>


<dt>parse</dt>
<dd><p>If <code>TRUE</code>, the labels will be parsed into expressions and
displayed as described in <code><a href="https://rdrr.io/r/grDevices/plotmath.html" class="external-link">?plotmath</a></code>.</p></dd>


<dt>check_overlap</dt>
<dd><p>If <code>TRUE</code>, text that overlaps previous text in the
same layer will not be plotted. <code>check_overlap</code> happens at draw time and in
the order of the data. Therefore data should be arranged by the label
column before calling <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text()</a></code>. Note that this argument is not
supported by <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_label()</a></code>.</p></dd>


<dt>na.rm</dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>


<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>


<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>


<dt>hjust</dt>
<dd><p>Moves the text horizontally</p></dd>


<dt>vjust</dt>
<dd><p>Moves the text vertically</p></dd>


<dt>angle</dt>
<dd><p>Defines the angle in which the text will be placed. *Note</p></dd>


<dt>nudge_y</dt>
<dd><p>Moves the text vertically an entire contig/sequence.
(e.g. <code>nudge_y = 1</code> places the text to the contig above)</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>These labeling functions use <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">ggplot2::geom_text()</a></code> under the hood.
Any changes to the aesthetics of the text can be performed in a ggplot2 manner.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># example data</span></span></span>
<span class="r-in"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">40</span>, <span class="fl">80</span>, <span class="fl">30</span>, <span class="fl">10</span>, <span class="fl">60</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">70</span>, <span class="fl">85</span>, <span class="fl">40</span>, <span class="fl">15</span>, <span class="fl">90</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>feat_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A1"</span>, <span class="st">"A2"</span>, <span class="st">"A3"</span>, <span class="st">"B1"</span>, <span class="st">"B2"</span>, <span class="st">"C1"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CDS"</span>, <span class="st">"CDS"</span>, <span class="st">"CDS"</span>, <span class="st">"CDS"</span>, <span class="st">"CDS"</span>, <span class="st">"CDS"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"geneA"</span>, <span class="st">"geneB"</span>, <span class="st">"geneC"</span>, <span class="st">"geneA"</span>, <span class="st">"geneC"</span>, <span class="st">"geneB"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">100</span>, <span class="fl">100</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">100</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># basic plot creation</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="gggenomes.html">gggenomes</a></span><span class="op">(</span>seqs<span class="op">=</span><span class="va">seqs</span>, genes<span class="op">=</span><span class="va">genes</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span><span class="fu"><a href="geom_bin_label.html">geom_bin_label</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span><span class="fu"><a href="geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># geom_..._text</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_text</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-1.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># geom_..._tag</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_tag</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-2.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># geom_..._note</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_note</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-3.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># with horizontal adjustment (`hjust`), vertical adjustment (`vjust`) </span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_text</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-4.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># using `nudge_y` and and `angle` adjustment</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_text</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">name</span><span class="op">)</span>, nudge_y<span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-5.png" alt="" width="960" height="480"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># labeling with manual input</span></span></span>
<span class="r-in"><span><span class="va">plot</span> <span class="op">+</span> <span class="fu">geom_gene_text</span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"This"</span>, <span class="st">"is"</span>, <span class="st">"an"</span>, <span class="st">"example"</span>, <span class="st">"test"</span>, <span class="st">"test"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="geom_feat_text-6.png" alt="" width="960" height="480"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thomas Hackl, Markus J. Ankenbrand, Bart van Adrichem.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

